md-dao/src/main/java/com/wipro/ats/bdre/md/dao/ProcessTemplateDAO.java

```diff
@@ -640,30 +640,30 @@ public void delete(Integer id) {
             session.beginTransaction();
 
             //select process_template_id from process_template where parent_process_id=p_p_id
-            Criteria fetchProcessTempId = session.createCriteria(ProcessTemplate.class).add(Restrictions.eq("processTemplate.processTemplateId", parentProcessId)).setProjection(Projections.property("processTemplateId"));
+            Criteria fetchProcessTempId = session.createCriteria(ProcessTemplate.class).add(Restrictions.eq(PARENTPROCESSTEMPLATEID, parentProcessId)).setProjection(Projections.property(PROCESSTEMPLATEID));
             List<Integer> processTempIdList = fetchProcessTempId.list();
             LOGGER.info("List size of process template id:" + fetchProcessTempId.list().size());
 
 
-            if (fetchProcessTempId.list().size() != 0) {
+            if (!fetchProcessTempId.list().isEmpty()) {
 
                 //select process_id from process where process_template_id in processTempIdList
-                Criteria fetchProcessId = session.createCriteria(com.wipro.ats.bdre.md.dao.jpa.Process.class).add(Restrictions.in("processTemplate.processTemplateId", processTempIdList)).setProjection(Projections.property("processId"));
+                Criteria fetchProcessId = session.createCriteria(com.wipro.ats.bdre.md.dao.jpa.Process.class).add(Restrictions.in(PARENTPROCESSTEMPLATEID, processTempIdList)).setProjection(Projections.property("processId"));
                 List<Integer> processIdList = fetchProcessId.list();
                 LOGGER.info("List size of process template id:" + fetchProcessId.list().size());
-                if (fetchProcessId.list().size() != 0) {
+                if (!fetchProcessId.list().isEmpty()) {
 
                     Criteria joinPropertiesAndProcess = session.createCriteria(com.wipro.ats.bdre.md.dao.jpa.Properties.class, "properties")
                             .createAlias("properties.process", "process")
                             .add(Restrictions.in("properties.process.processId", processIdList))//properties.process_id in processIdList
                             .add(Restrictions.eq("process.processId", processId));//process.process_id=p_id
-                    joinPropertiesAndProcess.setProjection(Projections.property("configGroup"));
+                    joinPropertiesAndProcess.setProjection(Projections.property(CONFIG_GROUP));
                     List<String> configGroupList = joinPropertiesAndProcess.list();
                     LOGGER.info("configGroupList size:" + joinPropertiesAndProcess.list().size());
 
-                    if (joinPropertiesAndProcess.list().size() != 0) {
+                    if (!joinPropertiesAndProcess.list().isEmpty()) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)